src/gromacs/nblib/interactions.h

```diff
@@ -46,18 +46,67 @@
 
 #include "gromacs/math/vectypes.h"
 
+#include <string>
+#include <vector>
+
 namespace nblib
 {
 
+using BondName = std::string;
+using EquiDist = real;
+using ForceConstant = real;
+
 //! Type of interaction
-struct HarmonicType
+struct HarmonicBondType
 {
     real equiDist;
     real forceConstant;
-    std::string atomIDi;
-    std::string atomIDj;
-    //int atomI;
-    //int atomJ;
+    BondName bondName;
+
+    HarmonicBondType(EquiDist ed, ForceConstant fc, BondName name)
+    : equiDist(ed), forceConstant(fc), bondName(std::move(name)) {}
+
+    //! Force explicit use of correct types
+    template<typename T, typename U, typename V>
+    HarmonicBondType(T ed, U fc, V name) = delete;
+};
+
+//! Type of interaction for storage
+struct Interaction
+{
+    void					 *interactionType;
+    size_t					  atomIndex1;
+    size_t					  atomIndex2;
+
+    Interaction (void* interactionPointer, size_t atomIndex1, size_t atomIndex2)
+    : interactionType(interactionPointer), atomIndex1(atomIndex1), atomIndex2(atomIndex2){}
+};
+
+struct InteractionContainer
+{
+public:
+    //! Specialization based on supported interaction types
+    template <class InteractionType>
+    void addInteraction(const InteractionType&	interactionType,
+                        const size_t atomIndex1,
+                        const size_t atomIndex2);
+
+    template <>
+    void addInteraction<HarmonicBondType>(const HarmonicBondType&	interactionType,
```

Comment:
Adding an interaction is something that should happen at the molecule level. The bond interface only should know about parameters related to bonds, but has no knowledge of atom indices or even of the concept of atom or AtomType.

Reply:
The `atomIndex` here is within the molecule level. If you see the client code, you'd understand. 

And struct `InteractionContainer` is a member of the `Molecule` class.